pkg/util/codec/codec.go

```diff
@@ -423,6 +424,178 @@ const (
 	KeepVarColumnLength
 )
 
+// PreAllocForSerializedKeyBuffer estimates key length
+func PreAllocForSerializedKeyBuffer(buildKeyIndex []int, chk *chunk.Chunk, tps []*types.FieldType, usedRows []int, filterVector []bool, nullVector []bool, serializeModes []SerializeMode, serializedKeysVectorBuffer [][]byte, memoryUsagePerRow *[]int64) error {
```

Comment:
Can you provide sufficient unit tests for such behavior?

Reply:
> Can you provide sufficient unit tests for such behavior?

Added now